token-syncer/integration/token_syncer/basic_test.clj

```diff
@@ -0,0 +1,411 @@
+;;
+;;       Copyright (c) 2017 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns token-syncer.basic-test
+  (:require [clojure.string :as str]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [plumbing.core :as pc]
+            [token-syncer.main :as main]
+            [token-syncer.syncer :as syncer])
+  (:import (java.util UUID)))
+
+(def basic-description {"cmd" "echo 'Hello World'", "cpus" 1, "mem" 2048, "metric-group" "syncer-test"})
+
+(defn- waiter-urls []
+  (let [waiter-uris (System/getenv "WAITER_URIS")]
+    (is waiter-uris)
+    (-> waiter-uris
+        (str/split #",")
+        sort)))
+
+(defn- waiter-api []
+  (main/init-waiter-api {:connection-timeout-ms 5000, :idle-timeout-ms 5000}))
+
+(deftest ^:integration test-environment
+  (log/info "****** Running: test-environment")
+  (testing "presence of environment variables"
+    (log/info "env.WAITER_URIS" (System/getenv "WAITER_URIS"))
+    (is (System/getenv "WAITER_URIS"))
+    (is (> (count (waiter-urls)) 1))))
+
+(defn- token->etag
+  "Retrieves the etag for a token on a waiter router."
+  [{:keys [load-token]} waiter-url token-name]
+  (-> (load-token waiter-url token-name)
+      (get :token-etag)
+      str))
+
+(defn- cleanup-token
+  [{:keys [hard-delete-token] :as waiter-api} waiter-urls token-name]
+  (log/info "Cleaning up token:" token-name)
+  (doseq [waiter-url waiter-urls]
+    (try
+      (let [token-etag (token->etag waiter-api waiter-url token-name)]
+        (hard-delete-token waiter-url token-name token-etag))
+      (catch Exception _))))
+
+(deftest ^:integration test-token-hard-delete
+  (testing "token sync hard-delete"
+    (let [waiter-urls (waiter-urls)
+          {:keys [load-token store-token] :as waiter-api} (waiter-api)
+          token-name (str "test-token-hard-delete-" (UUID/randomUUID))]
+      (try
+        (log/info "****** test-token-hard-delete ARRANGE")
```

Comment:
What's up with these log statements?

Reply:
Hmm, I'm not a big fan of the log lines. Can we do arrange-act-assert without these log lines?